src/core/lib/transport/metadata_batch.cc

```diff
@@ -116,13 +114,12 @@ static grpc_error* maybe_link_callout(grpc_metadata_batch* batch,
 
 static void maybe_unlink_callout(grpc_metadata_batch* batch,
                                  grpc_linked_mdelem* storage) {
-  grpc_metadata_batch_callouts_index idx =
-      GRPC_BATCH_INDEX_OF(GRPC_MDKEY(storage->md));
```

Comment:
can you think of a way to change the definition of `GRPC_BATCH_INDEX_OF` but getting the same behavior?

Reply:
Sure, one of the other things I tried out was to have GRPC_BATCH_INDEX_OF return an unsigned index which was in the range [0, u64max]. Then the following if statement (that you also commented on) would check to see if idx was > enum_max and return if so.

If I remember correctly using == instead of > didn't really change the code generated too much.

The question I do have is: is there a preference of changing GRPC_BATCH_INDEX_OF rather than what I've done here, and if so, why?

Reply:
So when would someone use GRPC_BATCH_INDEX_OF vs. do what you have here?